src/scripts/main.js

```diff
@@ -1,3 +1,248 @@
+  let callback;
+
+  switch (index) {
+    case 3: // age
+      callback = (a, b) => a.children[index].innerText
+      - b.children[index].innerText;
+      break;
+
+    case 4: // salary
+      callback = (a, b) => getSalary(a) - getSalary(b);
+      break;
+
+    default: // name && position && office
+      callback = (a, b) => {
+        return a.children[index].innerText
+          .localeCompare(b.children[index].innerText);
+      };
+  }
+
+  function getSalary(row) {
+    return +row.children[4].innerText.slice(1).split(',').join('');
+  }
+
+  // DESC or ASC order
+  if (lastChoise === index) {
+    x *= -1;
+  } else {
+    lastChoise = index;
+    x = 1;
+  }
+
+  tableBody.innerHTML = '';
+  list.sort((a, b) => callback(a, b) * x).forEach(tr => tableBody.append(tr));
+});
+
+// [2] focusing
+let activeRow;
+
+tableBody.addEventListener('click', action => {
+  if (activeRow) {
+    activeRow.classList.remove('active');
+  }
+
+  activeRow = action.target.closest('tr');
+  activeRow.classList.add('active');
+});
+
+// [3] adding form
+document.querySelector('table').insertAdjacentHTML('afterend', `
+<form class="new-employee-form">
+  <label>
+    Name: 
+    <input 
+      name="name" 
+      type="text"
+      data-qa="name"
+    >
+  </label>
+
+  <label>
+  Position: 
+  <input 
+    name="position" 
+    type="text"
+    data-qa="position"
+  >
+</label>
+
+<label>
+Office: 
+<select 
+  name="office" 
+  data-qa="office">
+  <option value="Tokyo">Tokyo</option>
+  <option value="Singapore">Singapore</option>
+  <option value="London">London</option>
+  <option value="New York">New York</option>
+  <option value="Edinburgh">Edinburgh</option>
+  <option value="San Francisco">San Francisco</option>
+  </select>
+</label>
+
+<label>
+Age: 
+<input 
+  name="age" 
+  type="number"
+  data-qa="age"
+>
+</label>
+
+<label>
+Salary: 
+<input 
+  name="salary" 
+  type="number"
+  data-qa="salary"
+>
+</label>
+<button 
+  type="submit">
+    Save to table
+</button>
+</form>`
+);
+
+const form = document.querySelector('.new-employee-form');
+
+// [4] converting salary
+function normilize(key, value) {
+  return key === 'salary'
+    ? `$${(+value).toLocaleString()}`
+    : value;
+}
+
+// [5] poping up notification
+function showNotification(type, message) {
+  form.insertAdjacentHTML('afterend', `
+  <div 
+    class="notification" 
+    data-qa="notification">
+      <h1 class="title"></h1>
+      <p></p>
+  </div>
+  `);
+
+  const notification = document.querySelector('.notification');
+  const notificationTitle = document.querySelector('.title');
+  const notificationDescription = document.querySelector('p');
+
+  switch (type) {
+    case 'error':
+      notification.classList.add('error');
+      notificationTitle.innerHTML = 'ERROR';
+      break;
+
+    case 'success':
+      notification.classList.add('success');
+      notificationTitle.innerHTML = 'SUCCESS';
+      break;
+
+    default:
+      return;
```

Comment:
delete this or replace `return` with `break`, bc in that case your notification won't be removed